packages/cloud_firestore/lib/src/firestore.dart

```diff
@@ -52,4 +60,64 @@ class Firestore {
     assert(path != null);
     return new DocumentReference._(this, path.split('/'));
   }
+
+  /// Runs a set of atomic database operations.
+  Future<Map<String, dynamic>> runTransaction(
+      TransactionHandler transactionHandler,
+      {Duration timeout: const Duration(seconds: 5)}) async {
+    assert(timeout.inMilliseconds > 0,
+        'Transaction timeout must be more than 0 milliseconds');
+    final int transactionId = _transactionHandlerId++;
+    _transactionHandlers[transactionId] = transactionHandler;
+    final Map<String, dynamic> result = await channel.invokeMethod(
+        'Firestore#runTransaction', <String, dynamic>{
+      'transactionId': transactionId,
+      'transactionTimeout': timeout.inMilliseconds
+    });
+    return result ?? <String, dynamic>{};
+  }
+}
+
+typedef Future<Map<String, dynamic>> TransactionHandler(Transaction tx);
```

Comment:
We might want to document the return value of a TransactionHandler here. I'm not sure it needs to be `Map<String, dynamic>`, the native equivalent of this can handle any serializable value (so maybe we should say `dynamic`).

I think it would be more Flutter style to call this argument Transaction even though it's more verbose. Though you may not need to name it here since it's redundant/obvious

Reply:
Named it `transaction`